### DISCUSSION TITLE
Delete Component thread after finishing a task

### DISCUSSION THREAD

**Discussion by sheershak-agarwal** [nasa/fprime#1037] [Q&A] [answered]
Hello,

I had a new implementation thought and wanted to clear out if that is possible. So in my deployer manager, I am deploying several component and one once I am done deploying I want to quit the deployer manager component thread. Currently my component is active. Is there a way to remove the component or quit its thread internally from the component?
Or are there other ways I can achieve this like by making the component passive?

Another side question, even if my active components are doing nothing, their threads are still active. Do they use any cpu resources? If yes then how much?

**Comment by LeStarch** [comment, marked as answer]
1. Active components with no messages to process do very little. This is because they are in an Os supplied sleep waiting on a queue message and are not scheduled to run until a message arrives.  It does use a bit of memory, but since flight software typically allocates memory up-front, we don't typically worry about de-allocating this.
2.  You may call <active component>.exit() to shutdown the thread if you really want to.  This is done in the reference topology when exiting. This will exit the thread's life cycle.  `.join` exists to, if you need to wait for it to exit. See: https://github.com/nasa/fprime/blob/567e2c547ab2526c9ba6fe10c8af9cda5954b3fb/Ref/Top/Topology.cpp#L276
3. You could by-hand instantiate an Os::Task and have it it return from the Task's function which could then be added to a passive component.
